ui/components/app/flask/snap-settings-card/snap-settings-card.js

```diff
@@ -0,0 +1,278 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+
+import Card from '../../../ui/card';
+import Box from '../../../ui/box';
+import IconWithFallback from '../../../ui/icon-with-fallback';
+import IconBorder from '../../../ui/icon-border';
+import Typography from '../../../ui/typography/typography';
+import ToggleButton from '../../../ui/toggle-button';
+import Chip from '../../../ui/chip';
+import ColorIndicator from '../../../ui/color-indicator';
+import Button from '../../../ui/button';
+
+import {
+  COLORS,
+  TYPOGRAPHY,
+  FONT_WEIGHT,
+  ALIGN_ITEMS,
+  DISPLAY,
+  TEXT_ALIGN,
+} from '../../../../helpers/constants/design-system';
+
+const STATUSES = {
+  INSTALLING: 'installing',
+  RUNNING: 'running',
+  STOPPED: 'stopped',
+  CRASHED: 'crashed',
+};
+
+const STATUS_COLORS = {
+  [STATUSES.INSTALLING]: COLORS.ALERT1,
+  [STATUSES.RUNNING]: COLORS.SUCCESS1,
+  [STATUSES.STOPPED]: COLORS.UI4,
+  [STATUSES.CRASHED]: COLORS.ERROR1,
+};
+
+const SnapSettingsCard = ({
+  name,
+  description,
+  icon,
+  dateAdded,
+  version,
+  url,
+  onToggle,
+  isEnabled = false,
+  onClick,
+  status,
+  className,
+  cardProps,
+  toggleButtonProps,
+  buttonProps,
+  chipProps,
+}) => {
+  const t = useI18nContext();
+  const [chipStatus, setChipStatus] = useState(STATUSES.INSTALLING);
+
+  const handleStatus = useCallback(() => {
+    switch (status) {
+      case STATUSES.INSTALLING: {
+        setChipStatus(STATUSES.INSTALLING);
+        break;
+      }
+      case STATUSES.RUNNING: {
+        setChipStatus(STATUSES.RUNNING);
+        break;
+      }
+      case STATUSES.STOPPED: {
+        setChipStatus(STATUSES.STOPPED);
+        break;
+      }
+      case STATUSES.CRASHED: {
+        setChipStatus(STATUSES.CRASHED);
+        break;
+      }
+      default: {
+        setChipStatus(STATUSES.INSTALLING);
+      }
+    }
+  }, [status]);
+
+  useEffect(() => {
+    handleStatus(status);
+  }, [handleStatus, status]);
```

Comment:
I think we can pass `status` to the chip and avoid trying and keep them in sync.

Reply:
Hmm, I think I agree with this -- seems simpler...